clang/lib/Frontend/FrontendActions.cpp

```diff
@@ -79,7 +79,7 @@ void ReadPCHAndPreprocessAction::ExecuteAction() {
   PP.EnterMainSourceFile();
   do {
     PP.Lex(Tok);
-  } while (Tok.isNot(tok::eof));
```

Comment:
The condition looks off to me. I’m in a phone so it’s a little hard to read, but does this read “read tokens while the token is not an end-of-file and the token is an end of repl”? Don’t we want it to keep reading each token that isn’t an end of file or isn’t an end of repl input?

Reply:
Yeah, it was a bad `!` 😅. Fixed.